core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -290,88 +312,113 @@ class AvroUpsertMonitorActor(val pollInterval: FiniteDuration,
-      val importFuture = for {
-        workspace <- dataSource.inTransaction { dataAccess =>
-          dataAccess.workspaceQuery.findByName(attributes.workspace).map {
-            case Some(workspace) => workspace
-            case None =>
-              throw new RawlsException(s"Workspace ${attributes.workspace} not found while importing entities")
-          }
-        }
-        petUserInfo <- getPetServiceAccountUserInfo(workspace.googleProjectId, attributes.userEmail)
-        importStatus <- importServiceDAO.getImportStatus(attributes.importId, attributes.workspace, petUserInfo)
-        _ <- importStatus match {
-          // Currently, there is only one upsert monitor thread - but if we decide to make more threads, we might
-          // end up with a race condition where multiple threads are attempting the same import / updating the status
-          // of the same import.
-          case Some(status) if status == ImportStatuses.ReadyForUpsert =>
-            publishMessageToUpdateImportStatus(attributes.importId, Option(status), ImportStatuses.Upserting, None)
-            toFutureTry(
-              initUpsert(attributes.upsertFile,
-                         attributes.importId,
-                         message.ackId,
-                         workspace,
-                         attributes.userEmail,
-                         attributes.isUpsert
-              )
-            ) map {
-              case Success(importUpsertResults) =>
-                val failureMessages = stringMessageFromFailures(importUpsertResults.failures, 100)
-                val baseMsg =
-                  s"Successfully updated ${importUpsertResults.successes} entities; ${importUpsertResults.failures.size} updates failed."
-                if (importUpsertResults.failures.isEmpty)
-                  publishMessageToUpdateImportStatus(attributes.importId,
-                                                     Option(status),
-                                                     ImportStatuses.Done,
-                                                     Option(baseMsg)
-                  )
-                else {
-                  val msg = baseMsg + s" First 100 failures are: $failureMessages"
-                  publishMessageToUpdateImportStatus(attributes.importId,
-                                                     Option(status),
-                                                     ImportStatuses.Error,
-                                                     Option(msg)
-                  )
-                }
-              case Failure(t) =>
-                val errMsg = t match {
-                  case errRpt: RawlsExceptionWithErrorReport => errRpt.errorReport.message
-                  case x                                     => x.getMessage
-                }
+  private def importEntities(message: PubSubMessage) = {
+    val attributes = new AvroUpsertMonitorMessageParser(message).parse
```

Comment:
this reverts back to pre-#2771 behavior: message parsing is now synchronous again and does not involve the database.